### DISCUSSION TITLE
Custom Scripts - Can't Get ObjectVar Variables to Work

### DISCUSSION THREAD

**Discussion by adparis99** [netbox-community/netbox#10605] [Q&A] [answered]
Hello, the issue is pretty much all there in the title. The script I'm trying to test is below, along with the error NetBox is outputing when I navigate to the scripts page. It seems to work fine with any of the other built-in variables like TextVar, StringVar, etc. I've tried changing the model to various others as well as using lower-case in case it was case sensitive in some way and am at a loss. Any thoughts?
      
```
        from extras.scripts import *
        class ConfigBuilder(Script):
            class Meta:
                name = "Generate Device Configuration"
                description = "Generates device configuration based on information stored in NetBox"
                commit_default = False

            test = ObjectVar(model = Device)
```

```
<class 'NameError'>

name 'Device' is not defined

Python version: 3.10.4
NetBox version: 3.3.0
```

**Comment by kkthxbye-code** [comment, marked as answer]
You need to import the Device model:

`from dcim.models import Device`

You can refer to the example at the bottom of the docs:

https://demo.netbox.dev/static/docs/customization/custom-scripts/

**Reply by adparis99** [reply]
Well I feel dumb. I've been using that link as a reference along with some other sites, but never scrolled all the way down. Thank you so much!

**Reply by adparis99** [reply]
If I could bug you with one more question, do you know if ObjectVar variables are iterable in the same way that objects are when accessed via pynetbox? I'm attempting to create dictionaries of device and interface properties for later use in the script but I'm coming up empty.

**Reply by kkthxbye-code** [reply]
MultiObjectVar returns a list, not sure if that is what you are looking for.

**Reply by candlerb** [reply]
There are also some examples in this repo: https://github.com/netbox-community/reports/tree/master/scripts
